Feature branch/micro-reddit

bini-i opened the pull request:
## The project has features of
- Creating a user
- Creating a post
- Making a comment under a post
#### It uses rails active records ORM(Object relational mapping) features, and it includes models, validations for the models and associations between the models.
## modifications
- added validations for the post and comment models

tnyandoro committed:
Getting started, creating the user column and db

bini-i committed:
Update schmema and User model

- add new migration add_user_name_to_users with not null constraint
- add validations to the User model for presence, uniqueness and length
constraint

bini-i committed:
Update User model

- add validation for uniqueness of email

tnyandoro committed:
Adding the association

bini-i committed:
Add Comment model

- with attribut of body, and references to User and Post model

bini-i committed:
Update Post and User model

- Add has_many associations to comments

tnyandoro committed:
Add the completed readme file

tnyandoro committed:
add the readme comments

tnyandoro committed:
Add the edieted readme

bini-i committed:
Add README.md

- add instuction for installation and running the rails project
- provide details of the project and the authors

bini-i committed:
Add git workflows

- added .github/workflows/linters.yml
- added .rubocop.yml

bini-i committed:
Update README.md

- added instruction for showing posts and comments of a user

peterrobert commented on app/models/post.rb:
```diff
@@ -0,0 +1,4 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+end
```
Very nice work adding the associations. However, kindly add validations.

- [ ]  Validate the body and the title

peterrobert commented on app/models/comment.rb:
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
```
Kindly add validation to the body.

peterrobert reviewed (changes requested):
### CHANGES REQUIRED :recycle: :stop_sign: 

HELLO @bini-i AND @tnyandoro :wave:  YOU HAVE DONE A GREAT JOB JUST A FEW CHANGES TO BE MADE AND YOUR PROJECT WILL BE APPROVED KINDLY CHECK BELOW :arrow_double_down: 

### if you have any question kindly reach me on slack @peterrobert

### HAPPY CODING :computer:

bini-i committed:
Update post.rb and comment.rb models

- add validation of presence for :title and :body attributes

bini-i committed:
Update README.md

- add preliminary data modeling section with diagramatic representaion
of the ER diagaram and mapping

trekab reviewed (approved):
### Approved! :tada: 
Congratulations team! You have done a good job attending to the requested changes from the last review. Keep up the good work!

_Happy coding!_ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)
